scylla/src/transport/locator/tablets.rs

```diff
@@ -0,0 +1,901 @@
+            ],
+        );
+        per_dc.insert(
+            "dc3".to_string(),
+            vec![(translator(Uuid::from_u64_pair(1, 3)).unwrap(), 1)],
+        );
+
+        assert_eq!(
+            replicas,
+            Ok(TabletReplicas {
+                all: replicas_uids
+                    .iter()
+                    .cloned()
+                    .map(|replica| (translator(replica).unwrap(), 1))
+                    .collect(),
+                per_dc
+            })
+        );
+    }
+
+    #[test]
+    fn table_tablets_empty() {
+        let tablets: TableTablets = TableTablets::new_for_test();
+        assert_eq!(tablets.tablet_for_token(Token::new(1)), None);
+    }
+
+    fn verify_ranges(tablets: &TableTablets, ranges: &[(i64, i64)]) {
+        let mut ranges_iter = ranges.iter();
+        for tablet in tablets.tablet_list.iter() {
+            let range = ranges_iter.next().unwrap();
+            assert_eq!(tablet.first_token.value(), range.0);
+            assert_eq!(tablet.last_token.value(), range.1);
+        }
+        assert_eq!(ranges_iter.next(), None)
+    }
+
+    fn insert_ranges(tablets: &mut TableTablets, ranges: &[(i64, i64)]) {
+        for (first, last) in ranges.iter() {
+            tablets.add_tablet(Tablet {
+                first_token: Token::new(*first),
+                last_token: Token::new(*last),
+                replicas: Default::default(),
+                failed: None,
+            });
+        }
+    }
+
+    #[test]
+    fn table_tablets_single() {
+        let mut tablets = TableTablets::new_for_test();
+
+        insert_ranges(&mut tablets, &[(-200, 1000)]);
+        verify_ranges(&tablets, &[(-200, 1000)]);
+
+        assert_eq!(
+            tablets.tablet_for_token(Token::new(-1)),
+            Some(&tablets.tablet_list[0])
+        );
+        assert_eq!(
+            tablets.tablet_for_token(Token::new(0)),
+            Some(&tablets.tablet_list[0])
+        );
+        assert_eq!(
+            tablets.tablet_for_token(Token::new(1)),
+            Some(&tablets.tablet_list[0])
+        );
+        assert_eq!(
+            tablets.tablet_for_token(Token::new(-200)),
+            Some(&tablets.tablet_list[0])
+        );
+        assert_eq!(tablets.tablet_for_token(Token::new(-201)), None);
+        assert_eq!(
+            tablets.tablet_for_token(Token::new(1000)),
+            Some(&tablets.tablet_list[0])
+        );
+        assert_eq!(tablets.tablet_for_token(Token::new(1001)), None);
+    }
+
+    #[test]
+    fn test_adding_tablets_non_overlapping() {
+        let mut tablets = TableTablets::new_for_test();
+        const RANGES: [(i64, i64); 5] = [
+            (-2000000, -1900001),
+            (-1900000, -1700001),
+            (-1700000, -1),
+            (0, 19),
+            (20, 10000),
+        ];
+
+        insert_ranges(&mut tablets, RANGES.as_slice());
+        verify_ranges(&tablets, RANGES.as_slice());
```

Comment:
Why not make `RANGES` a slice in the first place?

```suggestion
        const RANGES: &[(i64, i64)] = &[
            (-2000000, -1900001),
            (-1900000, -1700001),
            (-1700000, -1),
            (0, 19),
            (20, 10000),
        ];

        insert_ranges(&mut tablets, RANGES);
        verify_ranges(&tablets, RANGES);
```